src/ImageSharp.Drawing/Brushes/SolidBrush{TColor}.cs

```diff
@@ -72,13 +76,43 @@ public SolidBrushApplicator(TColor color)
             /// <returns>
             /// The color
             /// </returns>
-            public override TColor this[int x, int y] => this.color;
+            internal override TColor this[int x, int y] => this.color;
 
             /// <inheritdoc />
             public override void Dispose()
             {
                 // noop
             }
+
+            /// <inheritdoc />
+            internal override void Apply(float[] scanlineBuffer, int scanlineWidth, int offset, int x, int y)
+            {
+                DebugGuard.MustBeGreaterThanOrEqualTo(scanlineBuffer.Length, offset + scanlineWidth, nameof(scanlineWidth));
```

Comment:
I think it's better to use (non-debug) `Guard` here for more safety. `DebugGuard` is useful on :fire: :fire: execution paths, like indexer implementations.